File: apps/api/src/config/sports_config.py
# apps/api/src/config/sports_config.py

# Mapping frontend sport keys to The Odds API keys
SPORT_MAP = {
    "basketball_nba": "basketball_nba",
    "americanfootball_nfl": "americanfootball_nfl",
    "baseball_mlb": "baseball_mlb",
    "icehockey_nhl": "icehockey_nhl",
    "basketball_ncaab": "basketball_ncaab",
    "basketball_wnba": "basketball_wnba",
    "mma_mixed_martial_arts": "mma_mixed_martial_arts",
    "tennis_atp": "tennis_atp_french_open",
    "boxing_boxing": "boxing_boxing",
    "soccer_epl": "soccer_epl",
    "soccer_mls": "soccer_usa_mls",
}

# Mapping frontend sport keys to ESPN API paths
ESPN_MAP = {
    "basketball_nba": "basketball/nba",
    "americanfootball_nfl": "football/nfl",
    "baseball_mlb": "baseball/mlb",
    "icehockey_nhl": "hockey/nhl",
    "basketball_ncaab": "basketball/mens-college-basketball",
    "basketball_wnba": "basketball/wnba",
    "mma_mixed_martial_arts": "mma/ufc",
    "soccer_mls": "soccer/usa.1",
    "soccer_epl": "soccer/eng.1",
}

# Supported prop markets per sport as requested by user
PROP_MARKETS = {
    "basketball_nba": "player_points,player_rebounds,player_assists,player_threes,player_blocks,player_steals",
    "americanfootball_nfl": "player_pass_yds,player_rush_yds,player_reception_yds,player_tds,player_receptions",
    "baseball_mlb": "player_hits,player_home_runs,player_rbis,player_strikeouts,player_walks",
    "icehockey_nhl": "player_points,player_goals,player_assists,player_shots_on_goal",
    "basketball_ncaab": "player_points,player_rebounds,player_assists",
    "basketball_wnba": "player_points,player_rebounds,player_assists",
    "mma_mixed_martial_arts": "fighter_method_of_victory",
    "tennis_atp": "player_games_won",
    "boxing_boxing": "method_of_victory,total_rounds",
    "soccer_mls": "h2h,total_goals,both_teams_to_score",
    "soccer_epl": "h2h,total_goals,both_teams_to_score",
}

SPORT_DISPLAY = {
    "basketball_nba": "NBA",
    "americanfootball_nfl": "NFL",
    "baseball_mlb": "MLB",
    "icehockey_nhl": "NHL",
    "basketball_ncaab": "NCAAB",
    "basketball_wnba": "WNBA",
    "mma_mixed_martial_arts": "MMA/UFC",
    "tennis_atp": "Tennis",
    "boxing_boxing": "Boxing",
    "soccer_mls": "MLS",
    "soccer_epl": "EPL",
}

# ── Tiered polling intervals by sport priority ──
HIGH_FREQUENCY = [
    "basketball_nba",
    "americanfootball_nfl",
    "baseball_mlb",
    "icehockey_nhl",
    "basketball_ncaab",
    "americanfootball_ncaaf",
]

MEDIUM_FREQUENCY = [
    "soccer_usa_mls",
    "soccer_epl",
    "soccer_uefa_champs_league",
    "soccer_uefa_europa_league",
    "soccer_spain_la_liga",
    "soccer_germany_bundesliga",
    "soccer_italy_serie_a",
    "soccer_france_ligue_one",
    "mma_mixed_martial_arts",
    "tennis_atp_french_open",
    "tennis_wta_french_open",
    "tennis_atp_wimbledon",
    "tennis_wta_wimbledon",
    "tennis_atp_us_open",
    "tennis_wta_us_open",
    "tennis_atp_aus_open",
    "tennis_wta_aus_open",
]

LOW_FREQUENCY = [
    "golf_pga_championship",
    "golf_the_masters",
    "golf_us_open",
    "golf_the_open_championship",
    "boxing_boxing",
    "motorsport_formula_1",
    "aussierules_afl",
    "rugbyleague_nrl",
    "cricket_icc_world_cup",
]

ALL_SPORTS = list(set(HIGH_FREQUENCY + MEDIUM_FREQUENCY + LOW_FREQUENCY + list(SPORT_MAP.keys())))

File: apps/web/src/lib/sports.config.ts
export const SPORTS_CONFIG = {
    basketball_nba: { label: "NBA", icon: "🏀", season: true, espn: "nba", espn_path: "basketball/nba" },
    americanfootball_nfl: { label: "NFL", icon: "🏈", season: true, espn: "nfl", espn_path: "football/nfl" },
    baseball_mlb: { label: "MLB", icon: "⚾", season: true, espn: "mlb", espn_path: "baseball/mlb" },
    icehockey_nhl: { label: "NHL", icon: "🏒", season: true, espn: "nhl", espn_path: "hockey/nhl" },
    americanfootball_ncaaf: { label: "NCAAF", icon: "🏈", season: true, espn: "college-football", espn_path: "football/college-football" },
    basketball_ncaab: { label: "NCAAB", icon: "🏀", season: true, espn: "mens-college-basketball", espn_path: "basketball/mens-college-basketball" },
    basketball_wnba: { label: "WNBA", icon: "🏀", season: true, espn: "wnba", espn_path: "basketball/wnba" },
    soccer_epl: { label: "EPL", icon: "⚽", season: true, espn: "eng.1", espn_path: "soccer/eng.1" },
    soccer_uefa_champs_league: { label: "UCL", icon: "⚽", season: true, espn: "uefa.champions", espn_path: "soccer/uefa.champions" },
    mma_mixed_martial_arts: { label: "UFC", icon: "🥊", season: true, espn: "ufc", espn_path: "mma/ufc" },
    soccer_usa_mls: { label: "MLS", icon: "⚽", season: true, espn: "usa.1", espn_path: "soccer/usa.1" },
    tennis_atp: { label: "Tennis", icon: "🎾", season: true, espn: "atp", espn_path: "tennis/atp" },
    tennis_wta: { label: "Tennis (W)", icon: "🎾", season: true, espn: "wta", espn_path: "tennis/wta" },
    golf_pga: { label: "PGA", icon: "⛳", season: true, espn: "pga", espn_path: "golf/pga" },
    boxing_boxing: { label: "Boxing", icon: "🥊", season: true, espn: "boxing", espn_path: "boxing" },
    all: { label: "All Sports", icon: "🌍", season: true, espn: null, espn_path: "basketball/nba" },
} as const;

export type SportKey = keyof typeof SPORTS_CONFIG;
export const ALL_SPORTS = Object.keys(SPORTS_CONFIG) as SportKey[];
export const ACTIVE_SPORTS = ALL_SPORTS.filter(k => k !== 'all' && SPORTS_CONFIG[k].season);
export const DISPLAY_SPORTS = ALL_SPORTS.filter(k => k !== 'all');
